babel/spelling.py

```diff
@@ -0,0 +1,549 @@
+# -*- coding: utf-8 -*-
```

Comment:
This entire module redefines a lot of what I've seen at `babel/number_spelling.py`
Can you clean this a little? I suggest deleting some unused code and using a common module for duplicated code (e.g: the exceptions should reside at `babel/exceptions.py`)